homeassistant/components/imap/coordinator.py

```diff
@@ -166,8 +165,13 @@ def _decode_payload(part: Message) -> str:
             Falls back to the raw content part if decoding fails.
             """
             try:
-                return str(part.get_payload(decode=True).decode(self._charset))
+                decoded_payload: Any = part.get_payload(decode=True)
+                if TYPE_CHECKING:
+                    assert isinstance(decoded_payload, bytes)
+                content_charset = part.get_content_charset() or "utf-8"
+                return decoded_payload.decode(content_charset)
             except ValueError:
```

Comment:
What will raise ValueError?

Reply:
Invalid email content

Reply:
It is there because `base64` decoding could raise `InvalidBase64PaddingDefect` or `InvalidBase64CharactersDefect`, both sub classes of `ValueError`

[See: https://docs.python.org/3/library/email.compat32-message.html](https://docs.python.org/3/library/email.compat32-message.html#email.message.Message.get_payload)

Reply:
Ok.